crates/viewer/re_ui/src/notifications.rs

```diff
@@ -0,0 +1,393 @@
+    ///
+    /// Notifications are stored in order of ascending age, so the latest one is at the end.
+    data: Vec<Notification>,
+
+    has_unread_notifications: bool,
+
+    /// Panel that shows all notifications.
+    panel: NotificationPanel,
+
+    /// Toasts that show up for a short time.
+    toasts: Toasts,
+}
+
+impl Default for NotificationUi {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl NotificationUi {
+    pub fn new() -> Self {
+        Self {
+            data: Vec::new(),
+            has_unread_notifications: false,
+            panel: NotificationPanel::new(),
+            toasts: Toasts::new(),
+        }
+    }
+
+    pub fn has_unread_notifications(&self) -> bool {
+        self.has_unread_notifications
+    }
+
+    pub fn add(&mut self, level: Level, text: impl Into<String>) {
+        if !is_relevant(level) {
+            return;
+        }
+
+        self.data.push(Notification {
+            level: level.into(),
+            text: text.into(),
+
+            created_at: Zoned::now(),
+            ttl: base_ttl(),
+        });
+        self.has_unread_notifications = true;
+    }
+
+    pub fn success(&mut self, text: impl Into<String>) {
+        self.data.push(Notification {
+            level: NotificationLevel::Success,
+            text: text.into(),
+
+            created_at: Zoned::now(),
+            ttl: base_ttl(),
+        });
+    }
+
+    pub fn ui(&mut self, egui_ctx: &egui::Context, is_panel_visible: &mut bool) {
+        if *is_panel_visible {
+            let escape_pressed =
+                egui_ctx.input_mut(|i| i.consume_key(egui::Modifiers::NONE, egui::Key::Escape));
+            if escape_pressed {
+                *is_panel_visible = false;
+            }
+
+            self.has_unread_notifications = false;
+        }
+
+        self.panel.show(egui_ctx, &mut self.data, is_panel_visible);
+        self.toasts.show(egui_ctx, &mut self.data[..]);
+
+        if let Some(notification) = self.data.last() {
+            if !notification.ttl.is_zero() {
+                egui_ctx.request_repaint_after(notification.ttl);
+            }
+        }
+    }
+}
+
+struct NotificationPanel {
+    id: egui::Id,
+}
+
+impl NotificationPanel {
+    fn new() -> Self {
+        Self {
+            id: egui::Id::new("__notifications"),
+        }
+    }
+
+    fn show(
+        &self,
+        egui_ctx: &egui::Context,
+        notifications: &mut Vec<Notification>,
+        is_panel_visible: &mut bool,
+    ) {
+        if !*is_panel_visible {
+            return;
+        }
+
+        for notification in notifications.iter_mut() {
+            notification.ttl = Duration::ZERO;
+        }
+
+        let panel_width = 356.0;
+        let panel_max_height = 320.0;
+
+        let mut to_dismiss = None;
+
+        let notification_list = |ui: &mut egui::Ui| {
+            if notifications.is_empty() {
+                ui.label(
+                    egui::RichText::new("Nothing here!")
```

Comment:
I'm not sure about this copy. Maybe "No notifications"?

Reply:
It's right below the `Notifications` text, I don't like how it repeats the word twice so close together